### PR TITLE
`azurerm_spring_cloud_gateway_route_config` - order defaults to -1 to differenciate 0 and not set

### PR DESCRIPTION
**Pull request by ms-henglu**

### CODE REVIEW

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/springcloud/spring_cloud_gateway_route_config_resource.go [hunk 1]
```diff
@@ -92,6 +92,7 @@ func resourceSpringCloudGatewayRouteConfig() *pluginsdk.Resource {
 
 						"order": {
 							Type:     pluginsdk.TypeInt,
+							Default:  -1,
```
[the remark this answers is not in the document]
Comment (ms-henglu) [reply internal/services/springcloud/spring_cloud_gateway_route_config_resource.go:95]: Hi @katbyte , what do you think about this solution:

We add a `order_enabled` property, then only add the `order` in the payload when it's enabled?

[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply internal/services/springcloud/spring_cloud_gateway_route_config_resource.go:95]: let's just make this field Required which removes the problem?

[the remark this answers is not in the document]
Comment (ms-henglu) [reply internal/services/springcloud/spring_cloud_gateway_route_config_resource.go:95]: Makes sense, but it's still a breaking change, is it okay?

[the remark this answers is not in the document]
Comment (katbyte) [reply internal/services/springcloud/spring_cloud_gateway_route_config_resource.go:95]: @ms-henglu - that should be fine, just make sure there is a test if there is two of them as well checking the upgrade path for users

**Review by katbyte — APPROVED** [review]
LGTM  provided tests pass

**Pull request merged**